### ISSUE TITLE
Create 'dataset_iterator.py'

### ISSUE THREAD

**Issue by jeff1evesque** [jeff1evesque/earthquake-finder#75]
After #74 has been resolved, we will create `dataset_iterator.py`.  This file will be responsible for iterating the supplied dataset, and returning a `list` of `dict` records, of earthquakes within the specific parameters (i.e. radius, timeframe).  This file will `import` our `haversine.js` module to determine whether elements of the dataset is within the target `radius`.

This script will also implement _jsonschema_ validation on each iterated dataset.  The validation may look like the following:

``` python
  validate(json.loads(self.dataset)['xxx']['yyy'], jsonschema())
```

**Note:** the above validation implementation requires #81 to be resolved.

**Comment by jeff1evesque** [comment]
We need to implement our _radius-haversine_ logic, and validate each `dict` value we append to `self.target` within the `iterator` method.

**Comment by jeff1evesque** [comment]
4011c5f: was incorrectly committed to this issue, and has been removed:

```
git reset --hard HEAD~1
```

**Comment by jeff1evesque** [comment]
After we have resolved #74, we will need to import that file respectively to `dataset_iterator.py`.  The logic contained within `haversine.py` will be used within the `iterator` method to determine whether each data instance is within the specified `radius` of supplied `longitude`, and `latitude`.

**Comment by jeff1evesque** [comment]
7384ea7: was incorrectly committed to this issue, and has been removed:

```
git reset --hard HEAD~1
```
